### PR TITLE
feat(#1348): use common optimization for all tojos

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Create common optimisation for all tojos.

Closes: #1348

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@mximp Please, check

File: eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java [hunk 1]
```diff
@@ -228,19 +232,33 @@ private Optimization optimization(final Tojo tojo) {
...
+        } else {
+            opt = new OptTrain(opt, new ParsingTrain().empty());
+        }
+        return opt;
+    }
+
+    /**
+     * Optimization for specific tojo.
+     *
+     * @param tojo Tojp
```
[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java]: @mximp done

File: eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java [hunk 2]
```diff
@@ -210,12 +215,11 @@ private boolean optimizationRequired(final Tojo tojo) {
...
      *
-     * @param tojo Tojp
-     * @return Optimization for specific Tojo
+     * @return Optimization for all tojos.
      */
-    private Optimization optimization(final Tojo tojo) {
+    private Optimization common() {
         Optimization opt;
         if (this.trackOptimizationSteps) {
             opt = new OptSpy(targetDir.toPath().resolve(OptimizeMojo.STEPS));
```
[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java]: @mximp done

File: eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java [hunk 3]
```diff
@@ -154,9 +155,13 @@ public void exec() throws IOException {
      *  - <a href="https://stackoverflow.com/questions/49113207/completablefuture-forkjoinpool-set-class-loader/57551188#57551188">CompletableFuture / ForkJoinPool Set Class Loader</a>
      *  - <a href="https://stackoverflow.com/questions/74708979/is-there-any-difference-between-parallelstream-and-executorservice"> Difference between parallelStream() and ExecutorService</a>
      * @param tojo Tojo that should be optimized.
+     * @param common Optimization.
      * @return Optimization task.
      */
-    private Supplier<Integer> toOptimizationTask(final Tojo tojo) {
+    private Supplier<Integer> toOptimizationTask(
+        final Tojo tojo,
+        final Optimization common
```
[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java:163]: @mximp done

**Review by mximp — APPROVED** [review]

**Comment by volodya-lombrozo** [comment]
@yegor256 merge

**Comment by volodya-lombrozo** [comment]
@yegor256 reminder

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**